modules/cli/pages/actions-anonymization.adoc

```diff
@@ -0,0 +1,134 @@
+
+Sample configuration : 
+[source,yaml]
+----
+    arch_process_instance:
+      stringindex1:
+        actions:
+          - action: hash
+            value: SHA-256 
+----
+
+== Replace - Replace a value
+
+Replace the value with a specified value ( which can be empty ) 
+
+*Usage* : Value containing sensitive information you want to hide.
+
+*Parameters* :
+
+*value* : The value used to replace the current field. Default value empty
+
+Sample configuration :
+[source,yaml]
+----
+    arch_flownode_instance:
+      displayname:
+        actions:
+          - action: REPLACE
+            value: 'hidden'
+----
+
+== Replace_with_other - Replace a value by another column value of the same table
+
+Replace a value by another value of the same table and line.
+
+*Usage* : Value containing sensitive information you want to replace with another non sensitive meaningful value in the same table.
+
+*Parameters* :
+
+*value* : column name of the column to be use to replace the value.  No Default , mandatory
+
+Sample configuration :
+[source,yaml]
+----
+    arch_flownode_instance:
+      displayname:
+        actions:
+          - action: REPLACE_WITH_OTHER
+            value: name
+----
+== Regex_replace - Replace a value matching a regex expression
+
+Replace the value if the regexp match and use the matching group to create the new value.
+
+if multiple part of the value match , they are all replace. ( Behavior similar to  Matcher.replaceAll ) 
+
+If none of the regexp match , allow to configure a fallback which can be any of the other action. 
+
+*Usage* : Value containing some sensitive information with other non sensitive one and you want to keep the non sensitive information. 
+
+*Parameters* : 
+
+*pattern* : The regular expression to match  Default , mandatory
+
+*value* : The replacement pattern ( including captured group ) to used to change the value. No Default , mandatory
+
+*fallback* : action to apply if no regexp match.
+
+Sample configuration : 
+[source,yaml]
+----
+    arch_process_comment:
+      content:
+        actions:
+          - action: regex_replace
+            pattern: contract (\d+) is ready for user (\S+)\.(\S+)
+            value: contract XXXX is ready for $2
+          - action: regex_replace
+            pattern: The task Allocate repair agent on car (\S+) (is now assigned to .*)
+            value: The task Allocate repair agent on car *** $2
+        fallback:
+          - action: replace
+            value: hidden comment
+----
+
+== Keep - Keep a value
+
+Keep the value, no anonymization done. 
+
+*Parameters*: none
+
+Sample configuration :
+[source,yaml]
+----
+    arch_flownode_instance:
+      displayname:
+        actions:
+          - action: KEEP
+----
+
+== Remove_line - Remove a full line
+
+Remove the whole data line ( only possible on data contract and comment ) 
+
+*Parameters* : 
+optional where clause expressed as a regex to match with the value for the configured column. 
```

Comment:
**thought:** the where clause applies to all actions, so we could document it and give examples in all documented actions
Its configuration also changed recently in PR 352, and it can now be used like this:
```yaml
      arch_contract_data:
        val:
          actions:
          - action: REMOVE_LINE
            where:
            - column: name
              regex: PurchasedLicenseInput\.bypassSysDate
            - column: name
              regex: PurchasedLicenseInput\.caseCounterStartDate
            - column: name
              regex: PurchasedLicenseInput\.description
            - column: name
              regex: PurchasedLicenseInput\.endDate
            - column: name
              regex: PurchasedLicenseInput\.name
            - column: name
              regex: PurchasedLicenseInput\.numberCases
            - etc. ...
```

Reply:
I've changed the configuration sample. 
For the rest, how do we document it properly the where clause? How can it be used outside of arch_contract_data? Because it's mostly useful for this table because of the keys.

Reply:
@mbource i think it misse some words about maxSize defined by default to 512 as global option. The max size allow to truncate a big text. MasSize could be define also on each action. But it really use for KEEP and Replace.